pkg/services/sqldb/deprovision.go

```diff
@@ -7,19 +7,39 @@ import (
 	"github.com/Azure/open-service-broker-azure/pkg/service"
 )
 
-func (s *serviceManager) GetDeprovisioner(
+func (a *allInOneManager) GetDeprovisioner(
 	service.Plan,
 ) (service.Deprovisioner, error) {
 	return service.NewDeprovisioner(
-		service.NewDeprovisioningStep("deleteARMDeployment", s.deleteARMDeployment),
+		service.NewDeprovisioningStep("deleteARMDeployment", a.deleteARMDeployment),
 		service.NewDeprovisioningStep(
-			"deleteMsSQLServerOrDatabase",
-			s.deleteMsSQLServerOrDatabase,
+			"deleteMsSQLServer",
+			a.deleteMsSQLServer,
 		),
 	)
 }
 
-func (s *serviceManager) deleteARMDeployment(
+func (v *vmOnlyManager) GetDeprovisioner(
+	service.Plan,
+) (service.Deprovisioner, error) {
+	return service.NewDeprovisioner(
+		service.NewDeprovisioningStep("deleteARMDeployment", v.deleteARMDeployment),
```

Comment:
This will still need a step for deleting the server (same as in the all-in-one scenario).